examples/interpreting_monte_carlo_tree_search.ipynb

```diff
@@ -0,0 +1,1100 @@
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def mcts(\n",
+    "    smiles: str,\n",
+    "    scoring_function: Callable[[list[str]], list[float]],\n",
+    "    n_rollout: int,\n",
+    "    max_atoms: int,\n",
+    "    prop_delta: float,\n",
+    "    min_atoms: int = 15,\n",
+    "    c_puct: int = 10,\n",
+    ") -> list[MCTSNode]:\n",
+    "    \"\"\"Runs the Monte Carlo Tree Search algorithm.\n",
+    "\n",
+    "    Parameters\n",
+    "    ----------\n",
+    "    smiles : str\n",
+    "        The SMILES of the molecule to perform the search on.\n",
+    "    scoring_function : function\n",
+    "        A function for scoring subgraph SMILES using a Chemprop model.\n",
+    "    n_rollout : int\n",
+    "        The number of MCTS rollouts to perform.\n",
+    "    max_atoms : int\n",
+    "        The maximum number of atoms allowed in an extracted rationale.\n",
+    "    prop_delta : float\n",
+    "        The minimum required property value for a satisfactory rationale.\n",
+    "    min_atoms : int\n",
+    "        The minimum number of atoms in a subgraph.\n",
+    "    c_puct : float\n",
+    "        The constant controlling the level of exploration.\n",
+    "\n",
+    "    Returns\n",
+    "    -------\n",
+    "    list\n",
+    "        A list of rationales each represented by a MCTSNode.\n",
+    "    \"\"\"\n",
+    "\n",
+    "    mol = Chem.MolFromSmiles(smiles)\n",
+    "\n",
+    "    clusters, atom_cls = find_clusters(mol)\n",
+    "    nei_cls = [0] * len(clusters)\n",
+    "    for i, cls in enumerate(clusters):\n",
+    "        nei_cls[i] = [nei for atom in cls for nei in atom_cls[atom]]\n",
+    "        nei_cls[i] = set(nei_cls[i]) - {i}\n",
+    "        clusters[i] = set(list(cls))\n",
+    "    for a in range(len(atom_cls)):\n",
+    "        atom_cls[a] = set(atom_cls[a])\n",
+    "\n",
+    "    root = MCTSNode(smiles, set(range(mol.GetNumAtoms())))\n",
+    "    state_map = {smiles: root}\n",
+    "    for _ in range(n_rollout):\n",
+    "        mcts_rollout(\n",
+    "            root,\n",
+    "            state_map,\n",
+    "            smiles,\n",
+    "            clusters,\n",
+    "            atom_cls,\n",
+    "            nei_cls,\n",
+    "            scoring_function,\n",
+    "            min_atoms=min_atoms,\n",
+    "            c_puct=c_puct,\n",
+    "        )\n",
+    "\n",
+    "    rationales = [\n",
+    "        node\n",
+    "        for _, node in state_map.items()\n",
+    "        if len(node.atoms) <= max_atoms and node.P >= prop_delta\n",
+    "    ]\n",
+    "\n",
+    "    return rationales"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Running interpretation"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# MCTS options\n",
+    "property_for_interpretation = \"lipo\"\n",
```

Comment:
This feels like less of a MCTS option as it is part of the definition of the scoring function. I suggest putting this line in the next cell before `property_id = ...`.

Reply:
Done!